### ISSUE TITLE
0.3.0 tests fail

### ISSUE THREAD

**Issue by sanpii** [loichyan/nerdfix#8]
Hello,

The tests fails with the latest release:

    $ cargo test
        Finished test [unoptimized + debuginfo] target(s) in 0.05s
         Running unittests src/main.rs (target/debug/deps/nerdfix-f43ad6ff20021f57)
    
    running 3 tests
    test icon::tests::icon_from_str ... ok
    test icon::tests::icon_to_str ... ok
    test parser::tests::parser ... ok
    
    test result: ok. 3 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
    
         Running tests/cli.rs (target/debug/deps/cli-69cee76f507f960a)
    
    running 3 tests
    test check_with_input ... ok
    test check_json ... ok
    test check ... FAILED
    
    failures:
    
    ---- check stdout ----
    thread 'check' panicked at 'Unexpected stdout, failed fn(var)
    
    command=`"/home/sanpi/projects/arch/nerdfix/src/nerdfix-0.3.0/target/debug/nerdfix" "check" "tests/test-data.txt"`
    code=0
    stdout=<84 lines total>
    ```
    \u{1b}[0m\u{1b}[1m\u{1b}[38;5;10mnote\u{1b}[0m\u{1b}[1m: Found obsolete icon U+F752\u{1b}[0m
      \u{1b}[0m\u{1b}[34m┌─\u{1b}[0m tests/test-data.txt:1:27
      \u{1b}[0m\u{1b}[34m│\u{1b}[0m
    \u{1b}[0m\u{1b}[34m1\u{1b}[0m \u{1b}[0m\u{1b}[34m│\u{1b}[0m nf-mdi-folder_multiple = \"\u{1b}[0m\u{1b}[32m\u{f752}\u{1b}[0m\"
      \u{1b}[0m\u{1b}[34m│\u{1b}[0m                           \u{1b}[0m\u{1b}[32m^\u{1b}[0m \u{1b}[0m\u{1b}[32mIcon \'nf-mdi-folder_multiple\' is marked as obsolete\u{1b}[0m
      \u{1b}[0m\u{1b}[34m│\u{1b}[0m
      \u{1b}[0m\u{1b}[34m=\u{1b}[0m You could replace it with:
            1. \u{f0253} U+F0253 nf-md-folder_multiple
            2. \u{f13d3} U+F13D3 nf-md-folder_star_multiple
            3. \u{f0254} U+F0254 nf-md-folder_multiple_image
            4. \u{f147e} U+F147E nf-md-folder_multiple_plus
    
    \u{1b}[0m\u{1b}[1m\u{1b}[38;5;10mnote\u{1b}[0m\u{1b}[1m: Found obsolete icon U+F719\u{1b}[0m
      \u{1b}[0m\u{1b}[34m┌─\u{1b}[0m tests/test-data.txt:4:29
      \u{1b}[0m\u{1b}[34m│\u{1b}[0m
    \u{1b}[0m\u{1b}[34m4\u{1b}[0m \u{1b}[0m\u{1b}[34m│\u{1b}[0m nf-mdi-file_document_box = \"\u{1b}[0m\u{1b}[32m\u{f719}\u{1b}[0m\"
      \u{1b}[0m\u{1b}[34m│\u{1b}[0m                             \u{1b}[0m\u{1b}[32m^\u{1b}[0m \u{1b}[0m\u{1b}[32mIcon \'nf-mdi-file_document_box\' is marked as obsolete\u{1b}[0m
      \u{1b}[0m\u{1b}[34m│\u{1b}[0m
      \u{1b}[0m\u{1b}[34m=\u{1b}[0m You could replace it with:
            1. \u{f0219} U+F0219 nf-md-file_document
    ```
    <24 lines omitted>
    ```
             2. \u{e65d} U+E65D nf-seti-git
             3. \u{f1d3} U+F1D3 nf-fa-git
             4. \u{e702} U+E702 nf-dev-git
    
    \u{1b}[0m\u{1b}[1m\u{1b}[38;5;10mnote\u{1b}[0m\u{1b}[1m: Found obsolete icon U+F9E8\u{1b}[0m
       \u{1b}[0m\u{1b}[34m┌─\u{1b}[0m tests/test-data.txt:14:15
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m
    \u{1b}[0m\u{1b}[34m14\u{1b}[0m \u{1b}[0m\u{1b}[34m│\u{1b}[0m nf-mdi-tab = \"\u{1b}[0m\u{1b}[32m裡\u{1b}[0m\"
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m               \u{1b}[0m\u{1b}[32m^^\u{1b}[0m \u{1b}[0m\u{1b}[32mIcon \'nf-mdi-tab\' is marked as obsolete\u{1b}[0m
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m
       \u{1b}[0m\u{1b}[34m=\u{1b}[0m You could replace it with:
             1. \u{f04e9} U+F04E9 nf-md-tab
             2. \u{f523} U+F523 nf-oct-tab
             3. \u{f04eb} U+F04EB nf-md-table
             4. \u{f0ba0} U+F0BA0 nf-md-gitlab
    
    \u{1b}[0m\u{1b}[1m\u{1b}[38;5;10mnote\u{1b}[0m\u{1b}[1m: Found obsolete icon U+FC0A\u{1b}[0m
       \u{1b}[0m\u{1b}[34m┌─\u{1b}[0m tests/test-data.txt:17:27
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m
    \u{1b}[0m\u{1b}[34m17\u{1b}[0m \u{1b}[0m\u{1b}[34m│\u{1b}[0m nf-mdi-rhombus_outline = \"\u{1b}[0m\u{1b}[32mﰊ\u{1b}[0m\"
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m                           \u{1b}[0m\u{1b}[32m^\u{1b}[0m \u{1b}[0m\u{1b}[32mIcon \'nf-mdi-rhombus_outline\' is marked as obsolete\u{1b}[0m
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m
       \u{1b}[0m\u{1b}[34m=\u{1b}[0m You could replace it with:
             1. \u{f070c} U+F070C nf-md-rhombus_outline
             2. \u{f0ba6} U+F0BA6 nf-md-help_rhombus_outline
             3. \u{f11cf} U+F11CF nf-md-alert_rhombus_outline
             4. \u{f14dd} U+F14DD nf-md-rhombus_split_outline
    
    \u{1b}[0m\u{1b}[1m\u{1b}[38;5;10mnote\u{1b}[0m\u{1b}[1m: Found obsolete icon U+F554\u{1b}[0m
       \u{1b}[0m\u{1b}[34m┌─\u{1b}[0m tests/test-data.txt:20:29
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m
    \u{1b}[0m\u{1b}[34m20\u{1b}[0m \u{1b}[0m\u{1b}[34m│\u{1b}[0m nf-mdi-arrow_right_thick = \"\u{1b}[0m\u{1b}[32m\u{f554}\u{1b}[0m\"
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m                             \u{1b}[0m\u{1b}[32m^\u{1b}[0m \u{1b}[0m\u{1b}[32mIcon \'nf-mdi-arrow_right_thick\' is marked as obsolete\u{1b}[0m
       \u{1b}[0m\u{1b}[34m│\u{1b}[0m
       \u{1b}[0m\u{1b}[34m=\u{1b}[0m You could replace it with:
             1. \u{f0055} U+F0055 nf-md-arrow_right_thick
             2. \u{f09c6} U+F09C6 nf-md-arrow_top_right_thick
             3. \u{f19b0} U+F19B0 nf-md-arrow_right_thin
             4. \u{f09ba} U+F09BA nf-md-arrow_bottom_right_thick
    
    ```
    
    stderr=""
    ', /rustc/84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc/library/core/src/ops/function.rs:250:5
    note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    
    
    failures:
        check
    
    test result: FAILED. 2 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.37s
    
    error: test failed, to rerun pass `--test cli`

**Comment by loichyan** [comment]
Thank you! I forgot to update stdout after updating the cheat sheet.

**Comment by CUexter** [comment]
is this fixed in the release yet ?

**Comment by loichyan** [comment]
> is this fixed in the release yet ?

@CUexter this is not a bug in the binary, just a mistake in the test :)

**Comment by CUexter** [comment]
Ah i see. The aur pull the release tag and run tests before it was build hence the problem. I will contact the aur author Thanks for your help

**Comment by REmerald** [comment]
@loichyan , maybe make 0.3.1 release?  
Or maybe you can somehow change the commit base of 0.3.0?

Even though the binaries are unaffected, the source code assets/downloads in the tag/release are, because the release is based on the old commit.

These source code archives are very important if you want to compile your own binary (for various reasons), but don't wanna use unstable git version.  
That's why many AUR packages use them, [including the `nerdfix` package](https://aur.archlinux.org/packages/nerdfix).

**Comment by loichyan** [comment]
> That's why many AUR packages use them, [including the `nerdfix` package](https://aur.archlinux.org/packages/nerdfix).

I was not aware of this issue. It has been fixed in v0.3.1, thanks!
